### ISSUE TITLE
Linking .NET Godot project files into iOS XCode project breaks build

### ISSUE THREAD

**Issue by konsuko** [godotengine/godot#86019]
### Tested versions

- Reproducible in 4.2.1 RC1

### System information

M1 MacBook Pro, macOS Sonoma 14.0, Xcode 15.0

### Issue description

When I follow the guide outlined under "Active development considerations": https://docs.godotengine.org/en/latest/tutorials/export/exporting_for_ios.html#active-development-considerations

This does not work for me when using Godot 4.2.1 RC1 .NET Version.

I follow the exact steps to reproduce, the build *does work* but as soon as it starts up on device, Godot crashes with
```
USER ERROR: AudioOutputUnitStart failed, code: -50
   at: start (drivers/coreaudio/audio_driver_coreaudio.cpp:251)
Godot Engine v4.2.1.rc1.mono.official.daeb1c729 - https://godotengine.org
USER ERROR: AudioOutputUnitStart failed, code: -50
   at: start (drivers/coreaudio/audio_driver_coreaudio.cpp:251)
```

### Steps to reproduce

New empty project with a single C# script attached to a node, printing `GD.Print("Hello, world!");` in `_Ready()`
Build project, export for iOS, follow the steps here: https://docs.godotengine.org/en/latest/tutorials/export/exporting_for_ios.html#active-development-considerations

Build for iOS, run on device, crash.

### Minimal reproduction project (MRP)

N/A

**Comment by bruvzg** [comment]
> Active development considerations

This is likely not compatible with .NET (and probably won't be compatible in the future), which require AOT compilation and involve a lot of addition custom export steps implemented in the `mono` module export plugin. Documentation should be updated.

> USER ERROR: AudioOutputUnitStart failed, code: -50
>   at: start (drivers/coreaudio/audio_driver_coreaudio.cpp:251)

Probably unrelated. Usually this error happens when running in a simulator and incompatible external microphone is connected and selected as default (it should not affect anything except audio recording).

**Comment by konsuko** [comment]
Thanks, I understand. Still, it would be really great if we could somehow improve the iteration times for testing on device for iOS/.NET. Currently, an empty project takes ~40 seconds to build and an additional 10-15 seconds to load on the iPhone, which is a problem for iterative testing.

Compared to this, testing our older GDScript only project on device takes about half as much time.
`dotnet publish` seems to be the culprit here, maybe we could find a way to avoid having to run that every single time?

**Comment by bruvzg** [comment]
> dotnet publish seems to be the culprit here, maybe we could find a way to avoid having to run that every single time?

That's what building .NET code, so it's can't be skipped if any of the .NET code is changed.

**Comment by CreatedBySeb** [comment]
Also reported on `v4.3.stable.mono.official [77dcf97]` in #100891. If these instructions are not expected to work this seems like a documentation issue

**Comment by Eddie-Hartman** [comment]
I agree that it's a documentation issue. Is there an alternative to having to fully re-export from a Mac though? I'd like to be able to make a change to a C# Godot project (like a string of text in an app), then be able to just have Xcloud do the build and publish. It seems like if you're using C# with Godot for iOS exports, then you have to manually open Godot in MacOS and export to an xcode project every time you want to push an update. Is that still the case?
